src/os/shared.rs

```diff
@@ -33,17 +33,17 @@ impl Iterator for KeyboardEvents {
 
 fn get_datalink_channel(
     interface: &NetworkInterface,
-) -> Result<Box<dyn DataLinkReceiver>, std::io::Error> {
+) -> Result<Box<dyn DataLinkReceiver>, MyError> {
     let mut config = Config::default();
     config.read_timeout = Some(time::Duration::new(1, 0));
 
     match datalink::channel(interface, config) {
         Ok(Ethernet(_tx, rx)) => Ok(rx),
-        Ok(_) => Err(std::io::Error::new(
-            ErrorKind::Other,
-            "Unsupported interface type",
-        )),
-        Err(e) => Err(e),
+        Ok(_) => {
+            let error = MyError::new(MyErrorKind::PermissionError("Please do something".to_string()));
```

Comment:
@imsnif What do you think about this approach? Here we can define our own kind of error along with error message.

Reply:
Hey @SerhiiBilyk, the conflicts appear because the CI merges your work with the current master. To reproduce this locally, you need the latest changes.

You will have to add this repo (as opposed to your fork) as a remote if you didn't do it already. See [here](https://help.github.com/en/github/using-git/adding-a-remote) for instructions. After that, you can either rebase your branch on top of the current master, or merge master with your branch.

I hope this helps!